base/memory/shared_memory_helper.cc

```diff
@@ -41,6 +45,22 @@ bool CreateAnonymousSharedMemory(const SharedMemoryCreateOptions& options,
   // It doesn't make sense to have a open-existing private piece of shmem
   DCHECK(!options.open_existing_deprecated);
 #endif  // defined(OS_LINUX)
+#if defined(ANDROID) && defined(CASTANETS)
```

Comment:
Let's clean up the complex define branches.
Current:

```
#if !defined(OS_ANDROID) || defined(CASTANETS)
bool CreateAnonymousSharedMemory(...) {
  ...
#if defined(ANDROID) && defined(CASTANETS)
  // android specific impl.
#endif
  ...
}

bool PrepareMapFile(...) {
  // original impl.
}
#endif // !defined(OS_ANDROID) || defined(CASTANETS)
```

Below is my suggestion.

```
#if !defined(OS_ANDROID)
bool CreateAnonymousSharedMemory(...) {
  // original impl.
}

bool PrepareMapFile(...) {
  // original impl.
}
#elif defined(ANDROID) && defined(CASTANETS)
bool CreateAnonymousSharedMemory(...) {
  // only android specifc impl here.
}
#endif // !defined(OS_ANDROID)
```